src/include/migraphx/op/scatternd_op.hpp

```diff
@@ -36,36 +37,71 @@ namespace op {
 template <class Derived>
 struct scatternd_op : op_name<Derived>
 {
+    /** Validate input shapes and return the correct output shape.  For Scatter ops, the output
+     * is the same shape as the data tensor (first input), but cast to a standard shape.
+     */
     shape compute_shape(std::vector<shape> inputs) const
     {
-        check_shapes{inputs, *this}.has(3);
-        auto r         = inputs.front().lens().size();
-        auto q         = inputs.at(1).lens().size();
-        auto k         = inputs.at(1).lens().back();
-        auto ind_lens  = inputs.at(1).lens();
-        auto upd_lens  = inputs.back().lens();
-        auto data_lens = inputs.front().lens();
-        if(k > r)
-            MIGRAPHX_THROW("ScatterND: index of size " + std::to_string(k) +
-                           " is too large for tensor of rank " + std::to_string(r));
-        if(not(std::equal(ind_lens.begin(), ind_lens.begin() + q - 1, upd_lens.begin()) and
-               std::equal(data_lens.begin() + k, data_lens.end(), upd_lens.begin() + q - 1)))
-            MIGRAPHX_THROW("ScatterND: incorrect update shape. update.lens != indices.lens[0:q-1] "
-                           "++ data.lens[k:r-1]");
-        auto s = inputs.front();
-        if(s.broadcasted())
+        check_shapes{inputs, *this, true}.has(3);
+
+        if(migraphx::none_of(inputs, [](auto v) { return v.dynamic(); }))
         {
-            return {s.type(), s.lens()};
+            auto r         = inputs.front().lens().size();
+            auto q         = inputs.at(1).lens().size();
```

Comment:
Can use `shape.ndim()` here

Reply:
Done, also in a couple of other spots.  This had been awaiting the merge of ndim() to the develop branch.